fetch/api/abort/serviceworker-intercepted.https.html

```diff
@@ -100,6 +103,51 @@
     await promise_rejects_dom(t, "AbortError", w.DOMException, reader.read());
     await promise_rejects_dom(t, "AbortError", w.DOMException, reader.closed);
   }, "Stream errors once aborted.");
+
+  promise_test(async t => {
+    const scope = SCOPE + "?q=aborted-with-abort-reason";
+    await setupRegistration(t, scope, '../resources/sw-intercept.js');
+    const iframe = await with_iframe(scope);
+    add_completion_callback(_ => iframe.remove());
+    const w = iframe.contentWindow;
+
+    const controller = new w.AbortController();
+    const signal = controller.signal;
+
+    const fetchPromise = await w.fetch('data.json', { signal });
+
+    controller.abort(error1);
+
+    await promise_rejects_exactly(t, error1, fetchPromise);
+  }, "fetch() rejects with abort reason");
+
+  promise_test(async t => {
+    const scope = SCOPE + "?q=service-worker-observes-abort-reason";
+    await setupRegistration(t, scope, '../resources/sw-intercept-abort.js');
+    const iframe = await with_iframe(scope);
+    add_completion_callback(_ => iframe.remove());
+    const w = iframe.contentWindow;
+
+    const controller = new w.AbortController();
+    const signal = controller.signal;
+
+    const totalMessages = 0;
+
+    w.navigator.serviceWorker.addEventListener('message', event => {
```

Comment:
As we discussed, I think we should wait for the first message before aborting a fetch.

Something like:

```js

const fetchPromise = w.fetch('data.json', { signal });
await new Promise(resolve => {
  w.navigator.serviceWorker.addEventListener('message, t.step_func(event => {
    assert_equals(event.data, "fetch event has arrived");
    resolve();
  }), {once: true});
});

controller.abort(error1);

await new Promise(resolve => {
  w.navigator.serviceWorker.addEventListener('message, t.step_func(event => {
    assert_equals(event.data.message, error1.message);
    resolve();
  }), {once: true});
});
```

Reply:
I see, thank you!